src/txn/txn_timestamp.c

```diff
@@ -946,6 +946,32 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     return (0);
 }
 
+/*
+ * __wt_txn_set_timestamp_commit --
+ *     Directly set the commit timestamp in a transaction, bypassing parsing logic. Prefer this to
+ *     @ref __wt_txn_set_timestamp when string parsing is a performance bottleneck.
+ */
+int
+__wt_txn_set_timestamp_commit(WT_SESSION_IMPL *session, wt_timestamp_t ts)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+
+    WT_TRET(__wt_txn_context_check(session, true));
+
+    if (ts != WT_TS_NONE)
+        WT_RET(__wt_txn_set_commit_timestamp(session, ts));
+
+    __wt_txn_publish_durable_timestamp(session);
+
+    conn = S2C(session);
+    if (FLD_ISSET(conn->log_flags, WT_CONN_LOG_DEBUG_MODE) &&
+      FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED) && !F_ISSET(conn, WT_CONN_RECOVERING))
+        WT_RET(__wt_txn_ts_log(session));
```

Comment:
Now that we have two of these clauses, might want to add `__wt_log_ts_set()` into `log_inline.h`. No strong feeling, but seems cleaner.

Reply:
I've changed the new enum-based API to use a common internal function for this functionality - I think this addresses the issue.